haddock-api/src/Haddock/Interface/AttachInstances.hs

```diff
@@ -194,7 +195,8 @@ instance Ord SName where
 -- For the benefit of the user (looks nice and predictable) and the
 -- tests (which prefer output to be deterministic).
 data SimpleType = SimpleType SName [SimpleType]
```

Comment:
Any reason why you're not using `!` and `{-# UNPACK #-}` for the constructors?

Reply:
No reason, it's been non-strict before so I thought I'd just leave it.

Reply:
Alright. :)